### PR TITLE
New trit encoding for Transaction types

### PR DESCRIPTION
**Pull request by wusyong**
Refactor all types of transaction fields and transaction itself to TritBuf. Also derive macros of Serialize and Desrialze for them. 
Bundles and their builder types will implement and test later.

### CODE REVIEW

File: bee-bundle/src/transaction.rs [hunk 1]
```diff
@@ -3,197 +3,119 @@ use crate::constants::{ADDRESS, BUNDLE_HASH, IOTA_SUPPLY, NONCE, PAYLOAD, TAG};
...
+        Self(TritBuf::zeros(PAYLOAD.trit_offset.length))
     }
 
     pub fn from_str(payload: &str) -> Self {
         assert!(payload.len() <= PAYLOAD.tryte_offset.length);
         assert!(payload.chars().all(|c| c.is_tryte()));
 
-        let mut trytes = [TRYTE_ZERO; PAYLOAD.tryte_offset.length];
+        let missing_trytes = PAYLOAD.tryte_offset.length - payload.len();
+        let payload = payload.to_owned() + &TRYTE_ZERO.to_string().repeat(missing_trytes);
```
Comment (Alex6323) [inline bee-bundle/src/transaction.rs]: please use map collect instead

Comment (Alex6323) [reply bee-bundle/src/transaction.rs]: scrap what I said in my previous comment.
best solution is to use something like this:
`std::iter::repeat(TRYTE_ZERO).take(missing_trytes).collect::<String>();`

another very minor thing (mostly a matter of taste I guess):
I would change the variable name 'missing_trytes' to 'num_missing_trytes' as this the information we are really storing, and not trytes themselves. You know what I mean?

File: bee-bundle/src/transaction.rs [hunk 2]
```diff
@@ -358,7 +300,6 @@ pub enum TransactionBuilderError {
     InvalidValue(i64),
 }
 
-#[derive(Default)]
```
Comment (thibault-martinez) [inline bee-bundle/src/transaction.rs]: Why are we removing this ?

Comment (wusyong) [reply bee-bundle/src/transaction.rs]: Don't we agree on removing Default trait?

Comment (thibault-martinez) [reply bee-bundle/src/transaction.rs]: On Transactions, not on Builders

File: bee-bundle/src/transaction.rs [hunk 3]
```diff
@@ -329,12 +272,11 @@ impl fmt::Debug for Transaction {
     }
 }
 
-#[derive(Default)]
```
Comment (thibault-martinez) [inline bee-bundle/src/transaction.rs]: Same, I wouldn't remove this one. This is just a container, not a Transaction.

**Pull request merged**